src/Aspire.Hosting.Docker/DockerComposeServiceExtensions.cs

```diff
@@ -98,7 +98,10 @@ public static string AsEnvironmentPlaceholder(this ParameterResource parameter,
         return dockerComposeService.Parent.AddEnvironmentVariable(
             env,
             description: $"Parameter {parameter.Name}",
-            defaultValue: parameter.Secret || parameter.Default is null ? null : parameter.Value,
+            defaultValue: parameter.Secret || parameter.Default is null ? null : 
```

Comment:
Don't set the defaultValue parameter here, instead, leave it null and when writing the env file, use the source to resolve the default value. If the source is a parameter then use this logic. That will allow deferred usage of parameter.Value and instead we can use parameter.GetValueAsync